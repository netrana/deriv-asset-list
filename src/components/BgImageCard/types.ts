export interface Props {
  title?: string;
  buttonText?: string;
  className?: string;
  handleClick: ()=> void;
}
