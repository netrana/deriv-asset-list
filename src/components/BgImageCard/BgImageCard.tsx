import React, { FC } from "react";
import classNames from "classnames";
import Button from "components/Button";

import styles from "./bgImageCard.module.scss";
import { Props } from "./types";

const BgImageCard: FC<Props> = (props) => {
  const {
    title,
    className = "",
    buttonText = "Browse Cars",
    handleClick,
  } = props;
  return (
    <div
      className={classNames({
        [styles.root]: true,
        [styles.carBrowse]: true,
        [className]: Boolean(classNames),
      })}
    >
      <div className={styles.title}>{title}</div>
      {buttonText || buttonText !== "" ? (
        <Button className={styles.browseButton} onClick={handleClick}>
          {buttonText}
        </Button>
      ) : null}
    </div>
  );
};

export default BgImageCard;
