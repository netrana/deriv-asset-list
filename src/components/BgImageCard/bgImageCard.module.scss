@import 'styles/globals.scss';

.root {
  min-height: 50vh;
  border-radius: 10px;
  background-image: url('../../assets/images/find-car.png');
  background-size: cover;
  background-repeat: no-repeat;
  text-align: left;
  padding: 40px 40px 0px 40px;
  margin-bottom: 10px;

  .title {
    font-size: 36px;
    line-height: 48px;
    font-weight: 400;
    color: white;
    width: 100%;
    margin-bottom: 20px;
  }

  .browseButton {
    color: rgba(0, 0, 0, 0.85);
    background: #ffffff;
    border-radius: 200px;
  }

  @media only screen and (max-width: 480px) {
    .title {
      font-size: 24px;
      line-height: 36px;
      width: 100%;
    }
  }
}

@media only screen and (min-width: 1200px) {
  .root {
    padding: 40px 95px 0px 40px;
  }
}

@media only screen and (max-width: 975px) {
  .root {
    height: 30vh;
  }
}

@media only screen and (max-width: 768px) {
  .root {
    height: 20vh;
    min-height: 260px;
  }
}
