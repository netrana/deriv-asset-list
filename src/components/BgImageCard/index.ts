import BgImageCard from './BgImageCard';

export default BgImageCard;