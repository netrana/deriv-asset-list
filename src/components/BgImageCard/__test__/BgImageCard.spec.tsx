import BgImageCard from "../BgImageCard";
import render from "react-test-renderer";

jest.mock("components/Button/Button", () => "Button");

describe("BgImageCard component test", () => {
  describe("snapshot", () => {
    it("should match snapshot", () => {
      const tree = render
        .create(<BgImageCard handleClick={jest.fn} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
