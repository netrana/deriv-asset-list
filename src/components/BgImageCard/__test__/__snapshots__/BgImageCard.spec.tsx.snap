// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BgImageCard component test snapshot should match snapshot 1`] = `
<div
  className="root carBrowse "
>
  <div
    className="title"
  />
  <Button
    className="browseButton"
    onClick={[Function]}
  >
    Browse Cars
  </Button>
</div>
`;
